src/gsy_e/gsy_e_core/simulation.py

```diff
@@ -70,66 +70,195 @@ def __init__(self):
         self.current_slot_str = ""
         self.current_slot_number = 0
 
+    def update(self, slot_no, slot_count, time_params, config):
+        """Update progress info according to the simulation progress."""
+        run_duration = (
+                now(tz=TIME_ZONE) - time_params.start_time -
+                duration(seconds=time_params.paused_time)
+        )
+
+        if gsy_e.constants.RUN_IN_REALTIME:
+            self.eta = None
+            self.percentage_completed = 0.0
+        else:
+            self.eta = (run_duration / (slot_no + 1) * slot_count) - run_duration
+            self.percentage_completed = (slot_no + 1) / slot_count * 100
+
+        self.elapsed_time = run_duration
+        self.current_slot_str = get_market_slot_time_str(slot_no, config)
+        self.next_slot_str = get_market_slot_time_str(
+            slot_no + 1, config)
+        self.current_slot_number = slot_no
+
+        log.warning("Slot %s of %s - (%.1f %%) %s elapsed, ETA: %s", slot_no, slot_count,
+                    self.percentage_completed, self.elapsed_time,
+                    self.eta)
+
+    def log_simulation_finished(self, paused_duration, config):
+        """Log that the simulation has finished."""
+        log.info(
+            "Run finished in %s%s / %.2fx real time",
+            self.elapsed_time,
+            f" ({paused_duration} paused)" if paused_duration else "",
+            config.sim_duration / (self.elapsed_time - paused_duration)
+        )
+
 
 @dataclass
-class SimulationStateParameters:
+class SimulationStatusManager:
     """State of the Simulation class."""
-    # pylint: disable=too-many-instance-attributes
     paused: bool = False
     pause_after: duration = None
-    slot_length_realtime: duration = None
-    incremental: bool = False
     timed_out: bool = False
     stopped: bool = False
-    paused_time: int = 0  # Time spent in paused state, in seconds
     sim_status = "initializing"
-    start_time: datetime = None
-    tick_time_counter: datetime = None
+    incremental: bool = False
 
+    @property
+    def status(self):
```

Comment:
I see you created this property, can it include the `initializing` , `finished` and the other statuses as well?